couler/core/states.py

```diff
@@ -22,6 +25,18 @@
     # set cleanup_proto_workflow to an empty function for compatibility
     cleanup_proto_workflow = lambda: None  # noqa: E731
 
+name_salt = StringGenerator(r"[\c\d]{8}")
+
+
+def default_workflow_name_salter(name):
+    # The maximum length of a workflow name derives from the 
+    # maximum k8s resource name length (workflows are custom resources).
+    return "{0}-{1}".format(spinalcase(name), name_salt.render())[:62]
+
+
+workflow_name_salter = default_workflow_name_salter
```

Comment:
These can be internal variables (start name with underscore)?

Reply:
agreed